presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SpatialJoinNode.java

```diff
@@ -43,30 +44,57 @@
      */
     private final List<Symbol> outputSymbols;
     private final Expression filter;
+    private final Optional<Symbol> leftPartitionSymbol;
+    private final Optional<Symbol> rightPartitionSymbol;
+    private final Optional<String> kdbTree;
```

Comment:
Given the `KDBTree` will be JSON serializable, you can define this variable as `KDBTree` for better readability and type safety.

Reply:
Ping